homeassistant/components/panasonic_viera/media_player.py

```diff
@@ -92,16 +107,61 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     host = config.get(CONF_HOST)
     remote = RemoteControl(host, port)
 
+    if remote._type == TV_TYPE_ENCRYPTED:
+        needs_pairing = True
+        if path.exists(PANASONIC_VIERA_CONFIG_FILE):
+            with open(PANASONIC_VIERA_CONFIG_FILE, "rb") as config_dict_file:
+                config_dict = pickle.load(config_dict_file)
+                if host in config_dict:
+                    id = config_dict[host]["id"]
+                    key = config_dict[host]["key"]
+
+                    remote = RemoteControl(host, port, app_id=id, encryption_key=key)
+                    needs_pairing = False
+
+        if needs_pairing:
+            configurator = hass.components.configurator
```

Comment:
The configurator is legacy and should not be used for new code. Please create a config flow instead.

We have a scaffolding script to help you get started with that.

https://developers.home-assistant.io/docs/creating_component_index